docs/reference/indices/upgrade.asciidoc

```diff
@@ -22,11 +22,29 @@ is lost, the request will continue in the background, and
 any new requests will block until the previous upgrade is complete.
 
 [float]
+[[upgrade-parameters]]
+==== Request Parameters
+
+The `upgrade` API accepts the following request parameters:
+
+[horizontal]
+`upgrade_only_ancient_segments`:: If true, only very old segments (from a
```

Comment:
Also can we make a private boolean method shouldUpgrade for the hairy boolean logic. Github review tool just broke so i cant put this comment where it belongs.